core/handler/decorator.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Decorator  function
+"""
+
+import functools
+from core.exceptions.invalid_parameter import InvalidParamException
+from core.exceptions.unknown_error import UnknownException
+from jsonschema import validate, ValidationError, SchemaError
+
+def decorate(f):

```

Comment:
decorate(f) -> exception_handler_decorator(f)

decorate is a very generic term. Can we rename it so that purpose of this decorator function is understood. something like exception_handler_decorator. Otherwise this looks very neat way of handling exceptions by decorating the functions.